### ISSUE TITLE
Document implementation choices

### ISSUE THREAD

**Issue by cluelessjoe** [material-components/material-components-web#1291]
hi

When looking at the code, I saw heights in pixels. [In the docs, it feels like heights should be more adaptable](https://material.io/guidelines/layout/units-measurements.html#units-measurements-density-independent-pixels-dp). Previously I thought having heights in pixels wasn't such a good practices. 

But you know way more than I do and so your choices are most likely way better than mine. However I don't get the reasoning behind them. It would be great if this was available.

[Kafka Design documentation](https://kafka.apache.org/documentation.html#design) is an excellent example of how great this can be, helping anyone understanding the choices behind the code and providing a lot of in depth knowledge.

**Comment by lynnjepsen** [comment]
The spec page you linked to does say to use px when writing CSS. Did you have some particular concern with px? Perhaps I can better answer the reasoning question if I know what you're particularly concerned with.

**Comment by cluelessjoe** [comment]
Hi

First of all I didn't intend to be rude or anything, I hope it didn't feel this way :$

Regarding pixels:
- at first I stumbled upon the "long text" or "zoom issue" in list. Looking at the code I saw height was in fixed pixels. I remember having read pixels shouldn't be used for height/width/font because it doesn't play well with zooming and the like.
- then, looking at the documentation, it felt like the actual value should depend on the context: fixed pixel height wouldn't fit. 

I'm most likely missing something, so I created the current issue to try to understand better.

Thanks again

**Comment by Garbee** [comment]
This looks like a duplicate of #1261 to me. The same source of issues for that problem are what they appear to be asking for more information on.

The main units are `dp` in the spec which translate to `px` on the web. And the `sp` units for the font sizes translate to `rem` units. More information on the units can be seen in the [unit measurements](https://material.io/guidelines/layout/units-measurements.html#units-measurements-density-independent-pixels-dp) of the Material Guidelines. `dp` are "Density-independent pixels" which you use "dp to display elements uniformly on screens with different densities." This means it is not tied to the *internal contents* but the screen's display density itself.

Scalable Pixels, or `sp` units, are the closest direct reference to `rem` units.

> The primary difference between an sp and a dp is that sp preserves a user's font settings. Users who have larger text settings for accessibility will see the font size matched to their text size preferences.

So, they are meant to use the configured font size to adapt the display appropriately.

Basically for the sizing to change, we either need to correlate `dp` to `rem` as the more accurate conversion of units or rethink what the scale should be on the web for the foundation. Which, as everything reads currently correlating `dp` to `rem` is a big stretch.

**Comment by amsheehan** [comment]
As garbee points out in the linked issue, this is more than just an engineering decision. Since we currently do not have the bandwidth to reconsider our entire approach to how we size foundational elements, we're going to close this. The linked issue remains open.
